subsys/bluetooth/services/ias/ias_client.c

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <init.h>
+#include <stdint.h>
+#include <zephyr.h>
+#include <sys/check.h>
+#include <logging/log.h>
+#include <bluetooth/gatt.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/services/ias.h>
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_IAS_CLIENT)
+#define LOG_MODULE_NAME bt_ias_client
+#include "../../common/log.h"
+
+#define IS_HANDLE_VALID(handle) ((handle) != 0x0000)
+
+struct bt_ias_client {
+	/* IHandle for alert writes */
+	uint16_t write_handle;
+
+	/** Profile connection reference */
+	struct bt_conn *conn;
+
+	/* Gatt procedure parameters */
+	struct bt_uuid_16 uuid;
+	struct bt_gatt_discover_params discover;
+};
+
+static const struct bt_uuid *alert_lvl_uuid = BT_UUID_ALERT_LEVEL;
+static const struct bt_uuid *ias_uuid = BT_UUID_IAS;
+
+static const struct bt_ias_client_cb *ias_client_cb;
+static struct bt_ias_client client_list[CONFIG_BT_MAX_CONN];
+
+static struct bt_ias_client *client_by_conn(struct bt_conn *conn)
+{
+	return &client_list[bt_conn_index(conn)];
+}
+
+static void discover_failed(struct bt_conn *conn, int err)
+{
+	BT_DBG("conn %p", (void *)conn);
+
+	ias_client_cb->discover(conn, err);
+}
+
+static void client_reset(struct bt_ias_client *ias_client)
+{
+	bt_conn_unref(ias_client->conn);
+	(void)memset(ias_client, 0, sizeof(*ias_client));
+}
+
+int bt_ias_alert_write(struct bt_conn *conn, enum bt_ias_alert_lvl alert_lvl)
+{
+	int err;
+	uint8_t lvl = alert_lvl;
+
+	if (conn == NULL) {
+		return -ENOTCONN;
+	}
+
+	if (client_by_conn(conn)->write_handle == 0) {
+		return BT_GATT_ERR(BT_ATT_ERR_INVALID_HANDLE);
+	}
+
+	err = bt_gatt_write_without_response(conn,
+					client_by_conn(conn)->write_handle,
+					&lvl, sizeof(lvl), false);
+	if (err < 0) {
+		BT_ERR("IAS client alert level write failed: %d", err);
+		return err;
+	}
+
+	return err;
+}
+
+static uint8_t bt_ias_alert_lvl_disc_cb(struct bt_conn *conn,
+				const struct bt_gatt_attr *attr,
+				struct bt_gatt_discover_params *discover)
+{
+	struct bt_gatt_chrc *chrc = (struct bt_gatt_chrc *)attr->user_data;
+
+	if (attr == NULL) {
+		ias_client_cb->discover(conn, -BT_ATT_ERR_ATTRIBUTE_NOT_FOUND);
+		return BT_GATT_ERR(BT_ATT_ERR_ATTRIBUTE_NOT_FOUND);
```

Comment:
We generally keep an empty line between statements and `return`s
```suggestion
		ias_client_cb->discover(conn, -BT_ATT_ERR_ATTRIBUTE_NOT_FOUND);
		
		return BT_GATT_ERR(BT_ATT_ERR_ATTRIBUTE_NOT_FOUND);
```

Reply:
Done